### PR TITLE
Add support for OCI packaging

### PR DESCRIPTION
**Pull request by chouquette**

### CODE REVIEW

File: lib/omnibus/packagers/ociru.rb [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+require "omnibus/packagers/windows_base"
+require "fileutils"
+
+module Omnibus
+  class Packager::OCIRU < Packager::Base
+    id :ociru
+    intermediate_pkg_name = "package.tar.xz"
+
+    setup do
+    end
```
Comment (chouetz) [inline lib/omnibus/packagers/ociru.rb]: Is this needed?

File: lib/omnibus/packagers/ociru.rb [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+
+    build do
+      # create the payload directory, copy the install_dir and extra files to it
+      # TODO: does the makedirs do what we need in terms of file permissions?
```
Comment (chouetz) [inline lib/omnibus/packagers/ociru.rb]: should you use `create_sync_dir` then?

File: lib/omnibus/packagers/ociru.rb [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+          FileUtils.makedirs(destination)
+          FileUtils.cp(file, destination, preserve: true)
+        end
+      end
+      fl = filelist(payload_dir)
+
+      compress_env = { "XZ_OPT" => "-T#{compression_threads} -#{compression_level}" }
+
+      # create the archive
+      archive_file = windows_safe_path(staging_dir, intermediate_pkg_name)
```
Comment (chouetz) [inline lib/omnibus/packagers/ociru.rb:49]: pardon my lack of knowledge, is `windows_safe_path` to be used whatever the OS?

**Review by chouetz — APPROVED** [review]

File: lib/omnibus/packagers/ociru.rb (hunk 1 above)
Comment (chouquette) [reply lib/omnibus/packagers/ociru.rb]: I don't think it is. Removed.

File: lib/omnibus/packagers/ociru.rb (hunk 3 above)
Comment (chouquette) [reply lib/omnibus/packagers/ociru.rb:49]: yeah, it basically joins path using the correct slash based on the platform

File: lib/omnibus/packagers/ociru.rb (hunk 2 above)
Comment (chouquette) [reply lib/omnibus/packagers/ociru.rb]: indeed! Done

Comment (chouquette) [reply lib/omnibus/packagers/ociru.rb]: Ah now I remember! `FileSyncer.sync`already calls `create_sync_dir` so I just removed the `TODO`

**Review by bkabrda — APPROVED** [review]

**Pull request merged**